### ISSUE TITLE
Cannot set Precinct detective type for extensions that are not determined automatically

### ISSUE THREAD

**Issue by anicholls** [dependency-check-team/dependency-check#111]
I am trying to use `dependency-check` to look for missing and unnecessary dependencies in a CSS and Typescript + React component monorepo project. This has led me to using `precinct` for a detective. Unfortunately, I've discovered that precinct will not automatically use the correct detective for specific file types (`sass`, `less`, `scss`, `stylus`, `ts`, and `tsx`). You are required to pass this in explicitly as an option (e.g. `precinct(contents, {type: 'tsx'})`). Currently, `dependency-check` does not do this for you (or give you a way to do this).

### Example

When trying to check the dependencies of a `.tsx` file, inside `precinct`, either the [`walker.parse(content)`](https://github.com/dependents/node-precinct/blob/master/index.js#L49) call errors out and `[]` is returned for the dependencies, or it reaches their call to [`getModuleType.fromSource(ast)`](https://github.com/dependents/node-precinct/blob/master/index.js#L67), which can only return `es6`, `amd`, or `commonjs`. To correctly parse any of the file types listed above, you need to explicitly pass these as options to `precinct`. Since only the content of the file is passed to detectives/precinct (and not the extension/filename), there is no way to do this with upstream updates to precinct itself. Instead, some logic needs to be added to `dependency-check`.

### Potential Solution

Unfortunately, there's no way to do this statically from the command line as a CLI arg because you could be passing in a blob, in which case the extensions of the files being checked will be variable. I realize you've architected `dependency-check` to not reference the detectives directly, but I'm not sure if there's any way around it here. Something like the following added [here](https://github.com/dependency-check-team/dependency-check/blob/master/index.js#L326) would do the trick:

```js
const detectiveOptions = {};
if (
  detective.name === 'precinct' &&
  ['sass', 'less', 'scss', 'stylus', 'ts', 'tsx'].includes(ext.substr(1))
) {
  detectiveOptions['type'] = ext.substr(1);
}
const requires = detective(contents, detectiveOptions);
```

Let me know your thoughts on this, or any alternative approaches you have in mind. This is currently blocking the CI check I would like to get into place ASAP. I'd be happy to submit a PR once I have approval from a maintainer. Thanks!

**Comment by blakeembrey** [comment]
I think this is a similar request to https://github.com/dependency-check-team/dependency-check/issues/112, if we can get `precinct` to support a filename property it would work. They already do something similar for their `paperwork` function.

**Comment by voxpelli** [comment]
With the move to `precinct` as the default detective, custom support for this has been released in the `5.0.0-2` prerelease: https://github.com/dependency-check-team/dependency-check/releases/tag/v5.0.0-2

If one specifies `-e foo:precinct/foo` then `foo` will be used as the `type`. All default extensions has also been given correct such types.
